opm/output/eclipse/RestartValue.hpp

```diff
@@ -22,72 +22,58 @@
 #include <map>
 #include <vector>
 
-namespace Opm {
-
-
-    /*
-      Small convenience class used in the map passed to the
-      RestartIO::load( ) function. A class instance can be constructed
-      from a UnitSystem::measure value, which will default to a
-      required key, but it can also be constructed from a
-      two-parameter constructor, and then the required/not required
-      can controlled explicitly:
-
+#include <opm/parser/eclipse/Units/UnitSystem.hpp>
+#include <opm/output/data/Solution.hpp>
+#include <opm/output/data/Wells.hpp>
 
-         RestartIO::load(..., {{"PRESSURE" , UnitSystem::measure::pressure},
-                               {"MAX_SWAT" , {UnitSystem::measure::identity, false}} )
+namespace Opm {
 
-     The RestartKey( ) for pressure is implicitly created from
-     UnitSystem::measure::pressure and will be required, whereas the
-     MAX_SWAT keyword is optional.
 
-    */
     class RestartKey {
     public:
 
+        std::string key;
         UnitSystem::measure dim;
-        bool required = true;
+        bool required;
 
-        explicit RestartKey( UnitSystem::measure _dim)
-            : dim(_dim)
+        RestartKey( const std::string& _key, UnitSystem::measure _dim)
+            : key(_key),
+              dim(_dim),
+              required(true)
         {}
 
 
-        RestartKey( UnitSystem::measure _dim, bool _required)
-            : dim(_dim),
+        RestartKey( const std::string& _key, UnitSystem::measure _dim, bool _required)
+            : key(_key),
+              dim(_dim),
               required(_required)
         {}
 
     };
 
 
     /*
-      A simple struct - the only purpose is to facilitate return by value from the
-      RestartIO::load( ) function.
+      A simple class used to communicate values between the simulator and the
+      RestartIO function.
     */
 
 
-    struct RestartValue {
-
+    class RestartValue {
+    public:
+        using extra_vector = std::vector<std::pair<RestartKey, std::vector<double>>>;
```

Comment:
We should consistently use CamelCase for type names.

Reply:
Are you sure :-)

It has been removed now anyway - had exactly one point of use.